### ISSUE TITLE
Remove deprecated utils in t.python.util

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5516]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#5516|
|Type|task|
|Created|2012-03-07 19:45:44Z|
|Branch|https://github.com/twisted/twisted/tree/py-utils-5516|

`moduleMovedForSplit`, `dsu` ([#4339](https://github.com/twisted/twisted/issues/4339)) and `dict` from [twisted/python/util.py](twisted/python/util.py) are deprecated since 10.2 or earlier and can be removed.

<details><summary>Searchable metadata</summary>

```
trac-id__5516 5516
type__task task
reporter__thijs thijs
priority__low low
milestone__ 
branch__branches_py_utils_5516 branches/py-utils-5516
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1331149544000000 1331149544000000
changetime__1331392590000000 1331392590000000
version__None None
owner__thijs thijs
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra commented|
|-|-|

(In [33662]) Branching to 'py-utils-5516'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [33663]) remove deprecated utils, refs [#5516](https://github.com/twisted/twisted/issues/5516)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra removed owner|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/py-utils-5516).

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_5"></a>|@jesstess set owner to @thijstriemstra|
|-|-|

Thanks for cleaning up some style issues while working on this, thijs. Looks good, please merge!

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [33673]) Merge py-utils-5516: Remove `dsu`, `moduleMovedForSplit`, and `dict` from twisted.python.util (deprecated since 10.2).

Author: thijs
Reviewer: jesstess
Fixes: [#5516](https://github.com/twisted/twisted/issues/5516)
